libvast/vast/system/partition.hpp

```diff
@@ -14,6 +14,7 @@
 #pragma once
 
 #include "vast/aliases.hpp"
+#include "vast/chunk.hpp"
```

Comment:
We should only need a forward-declaration of `chunk_ptr` here, not the full header, so `vast/fwd.hpp` should suffice, and if it does not, this should be solved in `vast/fwd.hpp` instead.

The `vast/detail/stable_map.hpp` include can be removed I think.